common/src/KokkosFFT_utils.hpp

```diff
@@ -8,19 +8,45 @@
 #include <Kokkos_Core.hpp>
 #include <vector>
 #include <set>
+#include <string>
 #include <algorithm>
 #include <numeric>
 #include "KokkosFFT_traits.hpp"
 
+#ifndef KOKKOS_ENABLE_CXX17
+#include <source_location>
+#else
+#include <cstdlib>
+#endif
+
 namespace KokkosFFT {
 namespace Impl {
 
+inline void check_precondition(const bool expression, const std::string& msg) {
+  std::stringstream ss("file: ");
+#ifndef KOKKOS_ENABLE_CXX17
+  std::source_location location = std::source_location::current();
+  ss << location.file_name() << '(' << location.line() << ':'
+     << location.column() << ") `" << location.function_name() << "`: " << msg
+     << '\n';
+#else
+  ss << __FILE__ << '(' << __LINE__ << ") `" << __FUNCTION__ << "`: " << msg
+     << '\n';
+#endif
```

Comment:
This is always going to point to this file, not the calling site.  That is why I had suggested making the actual assertion a macro.

Reply:
Sorry, I thought they are equivalent. Fixed now